### PR TITLE
Ensure /etc/hosts has a header always - Fix conformance test

### PR DESCRIPTION
**Pull request by dghubble**
We have 2 scenarios where we copy /etc/hosts
- with host network (we just copy the /etc/hosts from node)
- without host network (create a fresh /etc/hosts from pod info)

We are having trouble figuring out whether a /etc/hosts in a
pod/container has been "fixed-up" or not. And whether we used
host network or a fresh /etc/hosts in the various ways we start
up the tests which are:

- VM/box against a remote cluster
- As a container inside the k8s cluster
- DIND scenario in CI where test runs inside a managed container

Please see previous mis-guided attempt to fix this problem at
ba20e63446815f8bcab4fc4c6e07eab2e2a9e121 In this commit we revert
the code from there as well.

So we should make sure:
- we always add a header if we touched the file
- we add slightly different headers so we can figure out if we used the
  host network or not.

Update the test case to inject /etc/hosts from node to another path
(/etc/hosts-original) as well and use that to compare.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Rework Kubelet set `/etc/hosts` behavior to fix conformance testability
```

### CODE REVIEW

**Comment by MaciekPytel** [comment]
@dghubble Does that fix any user visible problem, or is it just for fixing tests? If it's the former it needs release note. If the latter it probably still needs a release note, as the change is not limited to test code.

**Comment by dghubble** [comment]
Apologies, here's the background and I've added a note in the original description.

* In the past, the `KubeletManagedEtcHosts` test was known to fail in DIND and self-hosted style setups (see #60938, #61405, #60939, #60938). The failure was due the way the test was performed (originally https://github.com/kubernetes/kubernetes/commit/ba20e63446815f8bcab4fc4c6e07eab2e2a9e121)
* #61613 reworked the `KubeletManagedEtcHosts` e2e test to resolve those testing problems. The change was merged, but missed v1.10.0 and hasn't been cherry-picked. As a result, v1.10.x self-hosted style clusters will always fail that test when running e2e conformance tests. Here, I'm proposing it be cherry-picked into a release so it gets picked up by tools (e.g. sonobuoy) that try to accurately reflect conformance.

Separately, I've reproduced that e2e `KubeletManagedEtcHosts` failure, cherry-picked #61613 into the test suit, run against the same cluster, and shown e2e tests complete and pass successfully.

**Comment by dghubble** [comment]
Proposed: `Rework Kubelet set `/etc/hosts` behavior to fix conformance testability`

**Comment by MaciekPytel** [comment]
Thanks for explanation @dghubble, now I understand it more. Approved for cherry-pick (+ added some labels so it can merge during freeze on 1.11). 

For automated cherry-picks the release note is taken from parent PR, not cherry-pick. @dims Are you ok with copying the release note to your original PR?

**Comment by dims** [comment]
@MaciekPytel  Done!

**Comment by MaciekPytel** [comment]
@dims Thanks! @dghubble It's approved from my side, but you still need OWNER approval.

**Comment by MaciekPytel** [comment]
/hold 
Freezing branch for 1.10.5

**Comment by dghubble** [comment]
Anything else needed on my end to make it in v1.10.6?

**Comment by feiskyer** [comment]
@dghubble  The pr still needs approval.

/assign @yujuhong

**Comment by feiskyer** [comment]
@vishh Could you help to approve the PR?

/assign @vishh

**Review by vishh — APPROVED** [review]
/approve

**Pull request merged**
